clang/lib/Sema/SemaExpr.cpp

```diff
@@ -12681,6 +12681,40 @@ QualType Sema::CheckAddressOfOperand(ExprResult &OrigOp, SourceLocation OpLoc) {
     if (isa<CXXDestructorDecl>(MD))
       Diag(OpLoc, diag::err_typecheck_addrof_dtor) << op->getSourceRange();
 
+    if (getASTContext().getTargetInfo().getTriple().getArch() ==
+            llvm::Triple::aarch64 &&
+        getLangOpts().PointerAuthCalls && MD->isVirtual()) {
```

Comment:
Checking PointerAuthCalls is fine, no need for the target check.  Assuming we don't want to check this more generally.  Do we not have a check for this for MSVC?

Reply:
clang doesn't have any problems compiling the code in the test case when the target is MSVC (i386-pc-win32) since it can emit variadic thunks I mentioned in the radar. If we are going to upstream this, should this be a warning instead of an error for targets that aren't arm64e? I don't think we can start rejecting code that currently works fine?

Reply:
We've done so before when it was clearly a bug that it was accepted.  I think we'd need some committee guidance about it, though.

Reply:
OK, I'll send a patch or an RFC to the list later.

Reply:
Don't worry about it, I wrote up a DR for the C++ committee.